src/implementation/Client/HTTPClient/HTTPClient.ts

```diff
@@ -57,17 +56,27 @@ export default class HTTPClient implements IClient {
       this.clientUrl = `${this.clientHost}:${this.clientPort}/v1.0`;
     }
 
-    this.client = fetch;
+    if(!HTTPClient.client) {
+      HTTPClient.client = fetch;
+    }
 
     // Add a custom agent so we can decide if we want to reuse connections or not
     // we use an agent so we can reuse an open connection, limiting handshake requirements
     // Note: when using an agent, we will encounter TCPWRAP since the connection doesn't get destroyed
     if (this.options.isKeepAlive) {
-      this.httpAgent = new http.Agent({ keepAlive: true, keepAliveMsecs: 30 * 1000 });
-      this.httpsAgent = new https.Agent({ keepAlive: true, keepAliveMsecs: 30 * 1000 });
+      if(!HTTPClient.httpAgent) {
```

Comment:
Do we need to make httpsAgent static as well? It is being passed as params to the fetch client.

Reply:
I tested without making it static - Issue didn't resolve.

Reply:
One possible reason is: 
https://stackoverflow.com/a/64208818/18848251 - It seems Agent helps keep connections open and when required, they can be resued.
But, with new Agent every time, a new pool is allocated - so effectively no reuse.

Reply:
Ok, Java SDK reference which is doing something similar - https://github.com/dapr/java-sdk/blob/master/sdk/src/main/java/io/dapr/client/DaprHttpBuilder.java

Nit: to simplify the multiple if statements.

Reply:
Improved if-else and tested with that.